ic-os/components/setupos-scripts/check-network.sh

```diff
@@ -168,45 +162,38 @@ function ping_ipv6_gateway() {
     echo " "
 }
 
-function assemble_nns_nodes_list() {
-    NNS_URL_STRING=$(/opt/ic/bin/fetch-property.sh --key=.nns.url --config=${DEPLOYMENT})
-    NNS_URL_LIST=$(echo $NNS_URL_STRING | sed 's@,@ @g')
-}
-
 function query_nns_nodes() {
     echo "* Querying NNS nodes..."
 
-    i=0
-    success=0
-    nodes=$(echo ${NNS_URL_LIST} | wc -w)
-    # At least one of the provided URLs needs to work.
-    verify=1
-    for url in $(echo $NNS_URL_LIST); do
+    nns_urls=$(get_config_value '.icos_settings.nns_urls')
+
+    local urls=($(echo ${nns_urls} | jq -r '.[]'))
```

Comment:
Done!

Reply:
Update: can't be a single jq statement as jq will output the URLs as a single string rather than as separate elements. But merging into one line was a nice refactor